docs/source/connecting/connecting.md

```diff
@@ -35,6 +35,12 @@ async fn main() -> Result<(), Box<dyn Error>> {
 After successfully connecting to some specified node the driver will fetch topology information about
 other nodes in this cluster and connect to them as well.
 
+Driver maintains this pool of connections and each connection is capable of handling multiple requests in parallel. Driver will also route the request to node / shard that actually owns the data (unless load balancing policy that you use doesn't support it).
+For those reasons, we recommend to use one instance of `Session` per application.
+Creating short-lived `Session`'s (e.g. `Session` per request) is extremely discouraged and will result in great performance penalties because creating a `Session` is a costly process - it requires estabilishing a lot of TCP connections.
+Creating many `Session`'s in one application (e.g. `Session` per thread / per Tokio taks) is also discouraged, because it wastes resources - as mentioned before, `Session` maintains connection pool itself and can handle parallel queries, so you would be holding a lot of connections unnecessarily.
+If you need to share `Session` to different threads / Tokio tasks etc. use `Arc<Session>` - all methods of `Session` take `&self`, so it doesn't hinder the functionality in any way. In order to avoid cost of `Arc`'s atomic operations, you can estabilish all your `Arc`'s in the beggining as the atomic operations only happend when cloning `Arc`, not when using it's inner object.
```

Comment:
`share 'Session' to different threads` -> `share 'Session' with different threads`

Reply:
`avoid cost` -> `avoid the cost`

Reply:
typo: `happend` -> `happen`

Reply:
`it's inner object` -> `its inner object`

Reply:
I'm not sure that the remark about avoiding `Arc` atomic operations is realistic. Many client programs will spawn new tasks (I bet most of the web frameworks based on tokio do this) and then you will have to clone the `Arc`. The only alternative I see is putting the session into a global - but then you would either have to use unsafe to access it, or some wrapper that introduces synchronization overhead (rwlock, once cell, lazy static etc.)

Reply:
Dropped this section